pkg/utils/kubernetes/highavailability.go

```diff
@@ -94,13 +100,13 @@ func GetTopologySpreadConstraints(
 	return topologySpreadConstraints
 }
 
-func minDomains(numberOfZones, maxReplicas int32) *int32 {
-	// If the maximum replica count is lower than the number of zones, then we only need to set 'minDomains' to
+func calculateMinDomains(numDomains, maxReplicas int32) *int32 {
+	// If the maximum replica count is lower than the number of zones (e.g. zone count), then we only need to set 'minDomains' to
 	// the number of replicas because there is no benefit of enforcing a further zone spread for additional replicas,
 	// e.g. when a rolling update is performed.
-	if maxReplicas < numberOfZones {
+	if maxReplicas < numDomains {
 		return ptr.To(maxReplicas)
 	}
 	// Return the number of zones otherwise because it's not possible to spread pods over more zones than there are available.
```

Comment:
Should we adapt the comments also to the more general use of the function, i.e. now it is not only for zones, but also for nodes?